redis/src/test/scala/zio/redis/ApiSpec.scala

```diff
@@ -39,10 +45,204 @@ object ApiSpec extends BaseSpec {
+          for {
+            value    <- uuid
+            key1      = "custom_key_1"
+            key2      = "custom_key_2"
+            _        <- set(key1, value, None, None, None)
+            _        <- set(key2, value, None, None, None)
+            response <- keys("*custom*")
+          } yield assert(response)(hasSameElements(Chunk(key1, key2)))
+        },
+        testM("unlink existing key") {
+          for {
+            key     <- uuid
+            value   <- uuid
+            _       <- set(key, value, None, None, None)
+            removed <- unlink(key, Nil)
+          } yield assert(removed)(equalTo(1L))
+        },
+        testM("touch two existing keys") {
+          for {
+            key1    <- uuid
+            value1  <- uuid
+            _       <- set(key1, value1, None, None, None)
+            key2    <- uuid
+            value2  <- uuid
+            _       <- set(key2, value2, None, None, None)
+            touched <- touch(key1, List(key2))
+          } yield assert(touched)(equalTo(2L))
+        },
+        testM("scan entries") {
+          for {
+            key             <- uuid
+            value           <- uuid
+            _               <- set(key, value, None, None, None)
+            scan            <- scan(0L, None, None, None)
+            (next, elements) = scan
+          } yield assert(next)(isNonEmptyString) && assert(elements)(isNonEmpty)
+        },
+        testM("fetch random key") {
+          for {
+            key       <- uuid
+            value     <- uuid
+            _         <- set(key, value, None, None, None)
+            allKeys   <- keys("*")
+            randomKey <- randomKey()
+          } yield assert(allKeys)(contains(randomKey.get))
+        } @@ ignore,
+        testM("dump followed by restore") {
+          for {
+            key      <- uuid
+            value    <- uuid
+            _        <- set(key, value, None, None, None)
+            dumped   <- dump(key)
+            _        <- del(key, Nil)
+            restore  <- restore(key, 0L, dumped, None, None, None, None).either
+            restored <- get(key)
+          } yield assert(restore)(isRight) && assert(restored)(isSome(equalTo(value)))
+        } @@ ignore,
+        suite("expiration")(
+          testM("pExpire followed by pTtl") {
+            for {
+              key   <- uuid
+              value <- uuid
+              _     <- set(key, value, None, None, None)
+              exp   <- pExpire(key, 100.millis)
+              ttl   <- pTtl(key).either
+            } yield assert(exp)(isTrue) && assert(ttl)(isRight)
+          },
+          testM("pExpireAt followed by get and pTtl") {
+            for {
+              key      <- uuid
+              value    <- uuid
+              timeout  <- instantOf(100L)
+              _        <- set(key, value, None, None, None)
+              exp      <- pExpireAt(key, timeout)
+              response <- get(key)
+              ttl      <- pTtl(key)
```

Comment:
Note that you're guarding the similar call with either in a test above. To mitigate the flakiness, I'd rather do this with 2 get calls and sleep in between. Assertion would be that the first get returns something, while the second returns none. Please update all other tests as well.